behave_reportportal/behave_agent.py

```diff
@@ -0,0 +1,255 @@
+            return
+        description = step.text or ""
+        self._step_id = self._rp.start_test_item(
+            name="[{keyword}]: {name}".format(
+                keyword=step.keyword, name=step.name
+            ),
+            start_time=timestamp(),
+            item_type="STEP",
+            parent_item_id=self._scenario_id,
+            code_ref=self._code_ref(step),
+            description=description + self._build_table_content(step.table),
+            **kwargs
+        )
+
+    @check_rp_enabled
+    def finish_step(self, context, step, **kwargs):
+        """Finish test in Report Portal."""
+        if self._cfg.step_based:
+            self._finish_step_step_based(step, **kwargs)
+            return
+        self._finish_step_scenario_based(step, **kwargs)
+
+    def _get_launch_attributes(self):
+        """Return launch attributes in the format supported by the rp."""
+        attributes = self._cfg.launch_attributes or []
+        system_attributes = get_launch_sys_attrs()
+        system_attributes["agent"] = "{}-{}".format(
+            self.agent_name, self.agent_version
+        )
+        return attributes + _dict_to_payload(system_attributes)
+
+    @staticmethod
+    def _build_table_content(table):
+        if not table:
+            return ""
+
+        pt = PrettyTable(field_names=table.headings)
+        [pt.add_row(row.cells) for row in table.rows]
+        return "\n" + pt.get_string()
+
+    def _finish_step_step_based(self, step, status=None, **kwargs):
+        if step.status.name == "failed":
+            self._log_step_exception(step, self._step_id)
+        self._rp.finish_test_item(
+            item_id=self._step_id,
+            end_time=timestamp(),
+            status=status or self.convert_to_rp_status(step.status.name),
+            **kwargs
+        )
+
+    def _finish_step_scenario_based(self, step, **kwargs):
+        self._rp.log(
+            item_id=self._scenario_id,
+            time=timestamp(),
+            message="[{keyword}]: {name}. {text}{table}".format(
+                keyword=step.keyword,
+                name=step.name,
+                text=step.text or "",
+                table=self._build_table_content(step.table),
+            ),
+            level="INFO",
+            **kwargs
+        )
+        if step.status.name == "failed":
+            self._log_step_exception(step, self._scenario_id)
+
+    def _log_step_exception(self, step, item_id):
+        if not step.exception:
+            return
+        self._rp.log(
+            item_id=item_id,
+            time=timestamp(),
+            level="ERROR",
+            message="Step [{keyword}]: {name} was finished with "
+            "exception:\n{exception}".format(
+                keyword=step.keyword,
+                name=step.name,
+                exception=", ".join(step.exception.args),
+            ),
+        )
+
+    @staticmethod
+    def _item_description(item):
+        if not item.description:
```

Comment:
Could you please re-work similar code blocks so they conform to direct conditions? Thanks.

```
if item.description:
    return "Description:\n{}".format("\n".join(item.description))
```

Reply:
Ok, I will update, but just for curiosity: what's the difference between this case and [this](https://github.com/reportportal/client-Python/blob/0e6e370badb1cb187bec455ed060309930b400f5/reportportal_client/service.py#L487)? 
I think example from client demonstrates how early return improve readability of golden path.

Reply:
"Early return" in the client does not improve readability. It allows implementing batching mechanism. In our case, there is no reason to write this way. And, finally, [this](https://github.com/iivanou/client-Python/blob/log_manager/reportportal_client/core/log_manager.py#L74) is another implementation of the batching mechanism from me.